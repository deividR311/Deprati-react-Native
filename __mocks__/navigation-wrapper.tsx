import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
