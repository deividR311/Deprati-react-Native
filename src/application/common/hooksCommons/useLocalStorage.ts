import { remove } from 'lodash';
import { useState, useEffect } from 'react';
import { saveStorage, getStorage } from '../../state-manager/storage/storage';

export const StatusEnum = {
  pending: 'pending',
  success: 'success',
  error: 'error'
};
type StatusEnumType = keyof typeof StatusEnum;

const useLocalStorage = (key, initialValue) => {
  const [status, setStatus] = useState<StatusEnumType>(StatusEnum.pending);
  // State to store our value
  // Pass initial state function to useState so logic is only executed once
  const [storedValue, setStoredValue] = useState(initialValue);

  useEffect(() => {
    getStorage(key, initialValue).then(storage => {
      setStoredValue(storage);
      setStatus(StatusEnum.success);
    });
  }, [key]);

  // Return a wrapped version of useState's setter function that ...
  // ... persists the new value to localStorage.
  const setValue = (value, keyUid?: string) => {
    try {
      // Allow value to be a function so we have same API as useState
      const valueToStore =
        value instanceof Function ? value(storedValue) : value;
      // Save state
      setStoredValue(valueToStore);
      // Save to local storage
      if (typeof saveStorage !== 'undefined') {
        saveStorage(keyUid ?? key, valueToStore);
      }
    } catch (error) {}
  };
  return [storedValue, setValue, status];
};

export default useLocalStorage;
