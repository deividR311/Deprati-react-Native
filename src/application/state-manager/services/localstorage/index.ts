export * from './interface';
export * from './localstorage.slice';
export * from './useLocalStorage';
