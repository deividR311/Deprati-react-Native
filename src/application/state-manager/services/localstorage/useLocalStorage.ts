import { useSelector } from 'react-redux';
import { useAppDispatch } from '../..';
import {
  InitialLocalStorage,
  LocalStorageHook,
  LocalStorageKey,
  LocalStorageParams,
  LocalStorageType
} from './interface';
import {
  localStorageStateSelector,
  readAndStorageInState as _readAndStorageInState,
  save as saveInStorageAndState
} from './localstorage.slice';

export const useLocalStorage = (): LocalStorageHook => {
  const dispatch = useAppDispatch();
  const localStorageData = useSelector(localStorageStateSelector);

  const save = async (values: Partial<LocalStorageType>) => {
    return await dispatch(saveInStorageAndState(values));
  };

  const remove = async (keys: LocalStorageKey[]) => {
    const values = keys.reduce(
      (acc, key) => ({ ...acc, [key]: InitialLocalStorage[key] }),
      {} as LocalStorageParams
    );
    return await dispatch(saveInStorageAndState(values));
  };

  const readAndStorageInState = async () => {
    return await dispatch(_readAndStorageInState());
  };

  const fullClean = async () => {
    return await dispatch(saveInStorageAndState(InitialLocalStorage));
  };

  return {
    localStorageData,
    save,
    remove,
    readAndStorageInState,
    fullClean
  };
};
