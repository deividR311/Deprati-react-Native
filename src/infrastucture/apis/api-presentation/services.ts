import apiHybris from './index';
import { COMPONENTS } from './endpoints';
import type { ComponentDtoRequest } from './services.dto';

export class HybrisCmsServices {
  static getComponents(payload: ComponentDtoRequest): Promise<any> {
    payload = { fields: 'FULL', ...payload };
    return apiHybris.get(COMPONENTS, { params: payload });
  }
}
