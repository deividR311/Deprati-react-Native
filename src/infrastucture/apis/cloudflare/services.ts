import { urlCloudflare } from './index';
import { THUMBNAIL, VIDEO } from './endpoints';
import type { ThumbnailDtoRequest } from './services.dto';
import queryString from 'query-string';

export class CloudflareServices {
  static getThumbnailUrl(
    videoId: string,
    payload: ThumbnailDtoRequest
  ): string {
    payload = { time: '1s', ...payload };
    const params = queryString.stringify(payload);
    return `${urlCloudflare}${THUMBNAIL(videoId)}?=${params}`;
  }

  static getVideoUrl(videoId: string): string {
    return `${urlCloudflare}${VIDEO(videoId)}`;
  }
}
