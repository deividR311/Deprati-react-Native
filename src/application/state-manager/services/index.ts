import { signUpSlice } from './signUp';
import { loadingPlpSlice, filterPlpSlice } from './plp';
import { ContactlessPaymentSlice } from './contactless-payment';
import {
  SAP_API,
  HYBRISS_API
} from '../../../infrastucture/apis/common/hybriss.api';
import { LocalStorageSlice } from './localstorage';
import { creditSlice } from './credit';
import { modalSlice } from './credit/modal.slice';
import { favoriteSlice } from './favorite/favorite.redux';
import { indicationsSlice } from './indications/indications.slice';
import { CheckoutSlice } from './checkout/checkout.slice';
import { listenerAccountSlice } from './listenerAccount/listenerAccount.redux';
import { creditBalanceSlice } from './credit/creditBalance.redux';
import { LocationGPSSlice } from './location/location.slice';
import { servicesCreditSlice } from './credit/servicesCredit.slice';
import { pdpSlice } from './pdp/pdp.slice';

export const reducer = {
  [LocalStorageSlice.name]: LocalStorageSlice.reducer,
  [LocationGPSSlice.name]: LocationGPSSlice.reducer,
  [signUpSlice.name]: signUpSlice.reducer,
  [filterPlpSlice.name]: filterPlpSlice.reducer,
  [loadingPlpSlice.name]: loadingPlpSlice.reducer,
  [creditSlice.name]: creditSlice.reducer,
  [modalSlice.name]: modalSlice.reducer,
  [ContactlessPaymentSlice.name]: ContactlessPaymentSlice.reducer,
  [favoriteSlice.name]: favoriteSlice.reducer,
  [indicationsSlice.name]: indicationsSlice.reducer,
  [CheckoutSlice.name]: CheckoutSlice.reducer,
  [listenerAccountSlice.name]: listenerAccountSlice.reducer,
  [creditBalanceSlice.name]: creditBalanceSlice.reducer,
  [servicesCreditSlice.name]: servicesCreditSlice.reducer,
  [pdpSlice.name]: pdpSlice.reducer,
  [HYBRISS_API.reducerPath]: HYBRISS_API.reducer,
  [SAP_API.reducerPath]: SAP_API.reducer
};

export const apiMiddleware = [SAP_API.middleware, HYBRISS_API.middleware];
