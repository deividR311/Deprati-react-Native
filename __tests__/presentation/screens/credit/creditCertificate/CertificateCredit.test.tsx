import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../../__mocks__/navigation-wrapper';
import { useLocalStorage } from '../../../../../src/application/state-manager/services/localstorage/useLocalStorage';
import { LocalStorageKey } from '../../../../../src/application/state-manager/services/localstorage';
import CertificateCredit from '../../../../../src/presentation/screens/Credit/MoreServicesScreen/content/CertificateCredit/CertificateCredit';

jest.mock(
  '../../../../../src/application/state-manager/services/localstorage/useLocalStorage'
);
let screenTest: any;
describe('renders correctly in Snapshot CertificateCredit', () => {
  it('renders correctly in Snapshot', () => {
    useLocalStorage.mockImplementation(() => ({
      save: jest.fn(),
      localStorageData: {
        [LocalStorageKey.IsLogin]: true,
        [LocalStorageKey.User]: { UserDocumentId: 999222258 },
        [LocalStorageKey.Whatsapp]: {
          message: 'Hola',
          phone: '+593979461435'
        },
        [LocalStorageKey.MessagesApp]: {
          Certificate_Commercial_Reference: {
            title: 'Este servicio será cobrado de tu crédito De Prati.',
            description: 'Costo: $1,00 + IVA'
          },
          Certificate_Credit_at_day: {
            title: 'Este servicio será cobrado de tu crédito De Prati.',
            description: 'Costo: $0,63 + IVA'
          }
        }
      }
    }));

    screenTest = render(<NavigationWrapper screen={CertificateCredit} />);
    expect(screenTest.toJSON()).toMatchSnapshot();
  });
});
