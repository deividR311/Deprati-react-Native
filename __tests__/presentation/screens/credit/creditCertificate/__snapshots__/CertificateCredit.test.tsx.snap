// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly in Snapshot CertificateCredit renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        titleFontFamily="Roboto-Medium"
        titleFontSize={20}
        titleFontWeight="500"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Array [
                Object {
                  "alignItems": "center",
                  "overflow": "hidden",
                  "paddingTop": 10,
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
            style={
              Object {
                "marginBottom": 80,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0.44,
                    "lineHeight": 24,
                    "marginBottom": 5,
                    "textAlign": "left",
                    "width": "90%",
                  }
                }
              >
                Selecciona el certificado que deseas solicitar:
              </Text>
              <View
                style={
                  Object {
                    "width": "93%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginTop": 10,
                      "paddingHorizontal": 10,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessibilityLiveRegion="polite"
                      accessibilityRole="radio"
                      accessibilityState={
                        Object {
                          "checked": false,
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 18,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "borderColor": "rgba(0, 0, 0, 0.54)",
                            "borderRadius": 10,
                            "borderWidth": 2,
                            "height": 20,
                            "margin": 8,
                            "width": 20,
                          }
                        }
                      />
                    </View>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Roboto-Regular",
                          },
                          undefined,
                        ]
                      }
                    >
                      Referencia comercial
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "marginTop": 10,
                      "paddingHorizontal": 10,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessibilityLiveRegion="polite"
                      accessibilityRole="radio"
                      accessibilityState={
                        Object {
                          "checked": false,
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 18,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "borderColor": "rgba(0, 0, 0, 0.54)",
                            "borderRadius": 10,
                            "borderWidth": 2,
                            "height": 20,
                            "margin": 8,
                            "width": 20,
                          }
                        }
                      />
                    </View>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Roboto-Regular",
                          },
                          undefined,
                        ]
                      }
                    >
                      Crédito al día
                    </Text>
                  </View>
                </View>
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "rgba(0, 0, 0, 0.12)",
                          "height": 0.5,
                        },
                        undefined,
                        Object {
                          "alignSelf": "center",
                          "height": 1,
                          "marginVertical": 12,
                          "width": "100%",
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      Object {
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 16,
                        "fontWeight": "600",
                        "letterSpacing": 0.15,
                        "lineHeight": 24,
                      }
                    }
                  >
                    Tu certificado será enviado a:
                  </Text>
                  <View
                    style={
                      Object {
                        "color": "#979797",
                        "marginTop": 15,
                      }
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 6,
                          },
                          Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "rgba(0, 0, 0, 0.26)",
                            "borderRadius": 4,
                            "borderWidth": 1,
                          },
                        ]
                      }
                      testID="text-input-outline"
                    />
                    <View>
                      <View
                        style={
                          Array [
                            Object {
                              "paddingBottom": 0,
                            },
                            Object {
                              "minHeight": 48,
                              "paddingTop": 8,
                            },
                          ]
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="none"
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                              ],
                              "zIndex": 3,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            pointerEvents="none"
                            style={
                              Object {
                                "backgroundColor": "#ffffff",
                                "bottom": 4,
                                "left": 10,
                                "maxHeight": 2,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 6,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                ],
                                "width": 12,
                              }
                            }
                          />
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              Object {
                                "backgroundColor": "#ffffff",
                                "color": "transparent",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 18,
                                "maxWidth": -28,
                                "opacity": 0,
                                "paddingHorizontal": 0,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 29,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                  Object {
                                    "scaleY": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            Correo electrónico
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            onLayout={[Function]}
                            style={
                              Object {
                                "color": "rgba(0, 0, 0, 0.54)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "opacity": 0,
                                "paddingHorizontal": 14,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 28,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            Correo electrónico
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              Object {
                                "color": "rgba(0, 0, 0, 0.26)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "opacity": 0,
                                "paddingHorizontal": 14,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 28,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            Correo electrónico
                          </Text>
                        </View>
                        <TextInput
                          accessibilityIgnoresInvertColors={true}
                          accessibilityLabel="email"
                          autoComplete="off"
                          editable={false}
                          keyboardType="email-address"
                          maxFontSizeMultiplier={1.5}
                          multiline={false}
                          nativeID="email"
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder=""
                          placeholderTextColor="#979797"
                          selectionColor="#979797"
                          style={
                            Array [
                              Object {
                                "flexGrow": 1,
                                "margin": 0,
                                "paddingHorizontal": 14,
                                "zIndex": 1,
                              },
                              Object {
                                "height": 40,
                              },
                              Object {
                                "paddingBottom": 0,
                                "paddingTop": 0,
                              },
                              Object {
                                "color": "rgba(0, 0, 0, 0.54)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "textAlign": "left",
                                "textAlignVertical": "center",
                              },
                              false,
                              Array [
                                Object {},
                              ],
                            ]
                          }
                          testID="email"
                          underlineColorAndroid="transparent"
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "marginTop": 20,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                      }
                    }
                  >
                    ¿Quieres actualizar este correo?
Comunícate con nuestro 
                    <Text
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#E31A2C",
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 16,
                        }
                      }
                    >
                      Servicio al Cliente
                    </Text>
                    .
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0.12)",
                        "height": 0.5,
                      },
                      undefined,
                      Array [
                        Object {
                          "alignSelf": "center",
                          "height": 1,
                          "marginVertical": 12,
                          "width": "100%",
                        },
                        Object {
                          "marginVertical": 27,
                        },
                      ],
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "marginTop": 15,
                    "width": "93%",
                  }
                }
              >
                <View
                  accessibilityLabel="view_"
                  accessible={true}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      },
                      undefined,
                    ]
                  }
                  testID="view_"
                >
                  <View
                    accessibilityLabel=""
                    accessibilityLiveRegion="polite"
                    accessibilityRole="checkbox"
                    accessibilityState={
                      Object {
                        "checked": false,
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 18,
                          "height": 36,
                          "padding": 6,
                          "width": 36,
                        },
                      ]
                    }
                    testID=""
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "transform": Array [
                            Object {
                              "scale": 1,
                            },
                          ],
                        }
                      }
                    >
                      <Text
                        accessibilityElementsHidden={true}
                        allowFontScaling={false}
                        importantForAccessibility="no-hide-descendants"
                        pointerEvents="none"
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontSize": 24,
                            },
                            Array [
                              Object {
                                "lineHeight": 24,
                                "transform": Array [
                                  Object {
                                    "scaleX": 1,
                                  },
                                ],
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                            ],
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        󰄱
                      </Text>
                      <View
                        style={
                          Array [
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            Object {
                              "alignItems": "center",
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "borderColor": "#000000",
                              "borderWidth": 0,
                              "height": 14,
                              "width": 14,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                  <Text
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {},
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "Roboto-Regular",
                          },
                          Object {
                            "color": "#000000",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0.25,
                            "lineHeight": 20,
                          },
                        ]
                      }
                    >
                      acceptThe
                    </Text>
                    <Text
                      onPress={[Function]}
                      style={
                        Array [
                          Object {
                            "fontFamily": "Roboto-Regular",
                            "marginHorizontal": 5,
                          },
                          Object {
                            "color": "#E31A2C",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0.25,
                            "lineHeight": 20,
                          },
                        ]
                      }
                    >
                      termsAndConditions
                    </Text>
                  </Text>
                </View>
              </View>
              <View
                accessibilityLabel="GENERAR DOCUMENTO"
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#F5F6F9",
                    "borderColor": "#D3D3D3",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "height": 40,
                    "justifyContent": "center",
                    "marginTop": 14,
                    "opacity": 1,
                    "paddingVertical": 10,
                    "width": "90%",
                  }
                }
                testID="GENERAR DOCUMENTO"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 14,
                        "fontWeight": "700",
                        "letterSpacing": 0.8,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "color": "#D3D3D3",
                        "opacity": 0.5,
                      },
                    ]
                  }
                >
                  GENERAR DOCUMENTO
                </Text>
              </View>
            </View>
          </RCTScrollView>
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <Modal
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          />
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
